Assets/Scripts/PlayerAvatar.cs

```diff
@@ -90,6 +97,19 @@ public class PlayerAvatar : MonoBehaviour {
         transform.Translate(rb.velocity * Time.deltaTime);
     }
 
+    protected void handleAttack() {
+        fireCooldown = fireCooldown - Time.deltaTime;
+
+        Vector3 projectilePos = transform.position;
+        if (Input.GetButton("Fire1") && fireCooldown <= 0)
+        {
+            // Don't spawn right on top of player
+            projectilePos += new Vector3(1.0f, 0.0f, 0.0f);
+            newProjectile = Instantiate(projectilePrefab, projectilePos, Quaternion.identity) as GameObject;

```

Comment:
do we need to save a reference to the newProjectile?

if so: should it be an list/array instead of overwriting each time a new one is created? 

(if there is a value in instantiating and tracking many of them, we could consider the "object pool" pattern https://en.wikipedia.org/wiki/Object_pool_pattern)

Reply:
I don't think we actually do. I had it here before I moved the update logic into its own script 👍